IO/Ex1.hpp

```diff
@@ -0,0 +1,34 @@
+
+#pragma once
+
+#include <fstream> 
+#include <string>
+#include <vector>
+
+
+class FileHandler {
+public:
+    explicit FileHandler(const std::string path, std::ios_base::openmode permissions); //Ctor
+    FileHandler(FileHandler &other); //CCtor non-const because that tellg/p are modifying 
+    FileHandler(FileHandler &&other) noexcept; //MCtor
+
+    FileHandler& operator=(FileHandler &other); //Copy Assignment non-const because that tellg/p are modifying
+    FileHandler& operator=(FileHandler &&other) noexcept; //Move Assignment
+
+    void ReOpen(std::ios_base::openmode permissions);//Closes and reopen the file with new permissions
```

Comment:
Why `reOpen()` gets only permissions?
What is the usecase for this method?

Reply:
it is for users that opened the file in one way and wants to check the kind of opening.
e.g in one place they only need the read permissions and later on they need the write too .